lib/extensions/as_dependencies_interlock.rb

```diff
@@ -0,0 +1,20 @@
+class AsDependenciesInterlock
+  # The classic autoloader needs the ActiveSupport::Dependencies.interlock
+  # to avoid deadlocks.  With zeitwerk, this actually causes deadlocks so
+  # we only use the interlock with the classic autoloader.
+  def self.loading(&block)
+    if Rails.application.config.autoloader == :zeitwerk
+      yield
```

Comment:
What I think we should do, since `:zeitwerk` here is the final goal and merging that config change signals the end of this work, is put a one-time warning here...something lke:

```suggestion
      warn ":zeitwerk is the configured autloloader.  This patched file should be remove: #{__FILE__}"
      yield
```

Reply:
Likely this should move into a one-time method :) , but you get the idea

Reply:
you could just put this warn at the bottom of this file
So it would print out once at load time

Reply:
if `Rails.application.config.autoloader` is available at that time, then yeah that's an even better idea.

Reply:
I think I pulled that off... see latest updates.